src/biogeochem/CNPhenologyMod.F90

```diff
@@ -1962,7 +1972,11 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
                ! the onset_counter would change from dt and you'd need to make
                ! changes to the offset subroutine below
 
-            else if (hui(p) >= gddmaturity(p) .or. idpp >= mxmat(ivt(p))) then
+            else if ((idpp >= mxmat(ivt(p)) .and. (ivt(p) == nmiscanthus .or. ivt(p) == nirrig_miscanthus .or. &
+                      ivt(p) == nswitchgrass .or. ivt(p) == nirrig_switchgrass)) .or. &
+                      ((hui(p) >= gddmaturity(p) .or. idpp >= mxmat(ivt(p))) .and. & 
+                      (ivt(p) /= nmiscanthus .and. ivt(p) /= nirrig_miscanthus .and. &
+                      ivt(p) /= nswitchgrass .and. ivt(p) /= nirrig_switchgrass))) then
```

Comment:
@yanyanchengHydro - you gave some explanation of this in your email to me:

> I also added one more commit at lines 1975-1979 in CNPhenologyMod.F90 to make switchgrass and miscanthus have a longer growing season than other crops. This commit does not make much difference for a global scale simulation, but does make big difference for a single point 100% switchgrass/miscanthus simulation.

My understanding of this code change is that the harvest date for miscanthus and switchgrass does not at all depend on gddmaturity. So gddmaturity is still used to set other phenological stage timing, but is not used to set the harvest date. Is that correct?

@danicalombardozzi also calling your attention to this change.

Reply:
Thanks for pointing this out @billsacks. @yanyanchengHydro, I have two related questions here. First, what is happening to the biofuel crops during the grain fill phase? Are they actually producing grain, or did you change parameters so that there is no grain? This also relates to our discussion related to  [#884 (comment)](https://github.com/ESCOMP/CTSM/pull/884#issuecomment-604101236). Second, from my understanding of the change you made, the biofuels grow for the same length of time everywhere. Would it make more sense to increase the gddmaturity parameter (or something similar) so that there is some climate-based spatial variation in growing season length? Are biofuels always harvested after a certain length of time?

Reply:
@danicalombardozzi @billsacks Thank you for your comments on this. I did not change grain parameters so they are producing grain. The growing season length is typically around six months. But I agree it should have climate-based spatial variations.
For increasing gddmaturity parameter, do you mean parameters like hybgdd? I tuned it earlier but did not achieve the longer growing season goal. But I will test it again. Do you have any other suggested parameters that I can test?

Reply:
Thanks for the clarification @yanyanchengHydro. I wonder how much grain is being accumulated for each biofuel type, and whether it is at all realistic. During the grain fill phase, allocation shifts away from the leaves and to the grain. It seems like it might be worth looking at the grain pool for these crops at the same time as exploring the growing season length options. 

In my above suggestion, the hybgdd parameter is what I thought might be useful to try to change. I wonder how dramatic the change in this would have to be to increase the growing season length. Two other thoughts that come to mind: I wonder if it's worth exploring making the leaf emergence phase longer, which will likely reduce the amount of grain produced (although I'm not sure this would make the growing season longer) -- I think this is the grnfill parameter. Another option is to create some sort of latitudinal dependence (we've done this for planting for some crops), although I don't think this is the best solution. 

Do you have a sense for whether the maximum LAI threshold is being hit, which would trigger the crops to enter into the grain fill phase earlier than the temperature triggers? If this is the case, then it might also be worth playing around with the allocation parameters.